.github/workflows/documentation.yml

```
@@ -0,0 +1,31 @@
+name: documentation
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
+
+jobs:
+
+  build:
+    name: Documentation for (${{ matrix.python-version }}, ${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ['ubuntu-latest']
+        python-version: ['3.9']
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/install-python-and-package
+        with:
+          python-version: ${{ matrix.python-version }}
```

```suggestion
    name: Build documentation
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
    steps:
      - uses: actions/checkout@v2
      - uses: ./.github/actions/install-python-and-package
```

Slight simplifications: 1) no need for a matrix anymore and 2) no need to specify Python version (the default argument is already 3.9).